### ISSUE TITLE
Using websockets.connect without async context manager is broken in 5.0

### ISSUE THREAD

**Issue by SimSimY** [aaugustin/websockets#411]
It looks like the new version breaks the ability to use the websockets connect object without async context even though the documentation [clearly specifies](https://websockets.readthedocs.io/en/stable/api.html?highlight=connect#websockets.client.connect) that this is possible:

> connect() returns an awaitable. Awaiting it yields an instance of WebSocketClientProtocol which can then be used to send and receive messages.
> 
> On Python ≥ 3.5.1, connect() can be used as a asynchronous context manager. In that case, the connection is closed when exiting the context. 

Here is an example showcasing the error:
the "server code" is simply the code from the getting started [basic example](https://websockets.readthedocs.io/en/4.0.1/intro.html#basic-example)
```python
#!/usr/bin/env python

import asyncio
import websockets

async def hello(websocket, path):
    name = await websocket.recv()
    print("< {}".format(name))

    greeting = "Hello {}!".format(name)
    await websocket.send(greeting)
    print("> {}".format(greeting))

start_server = websockets.serve(hello, 'localhost', 8765)

asyncio.get_event_loop().run_until_complete(start_server)
asyncio.get_event_loop().run_forever()

```

and the client:
```python
#!/usr/bin/env python

import asyncio
import websockets


async def hello():
    websocket = await websockets.connect('ws://localhost:8765')
    name = input("What's your name? ")
    await websocket.send(name)
    print("> {}".format(name))

    greeting = await websocket.recv()
    print("< {}".format(greeting))


asyncio.get_event_loop().run_until_complete(hello())

```

The code runs with version `~=4.0`,  but on `~=5.0` `TypeError` is thrown:

```python
Traceback (most recent call last):
  File "XXX/client.py", line 17, in <module>
    asyncio.get_event_loop().run_until_complete(hello())
  File "/usr/local/Cellar/python/3.6.4_4/Frameworks/Python.framework/Versions/3.6/lib/python3.6/asyncio/base_events.py", line 467, in run_until_complete
    return future.result()
  File "XXX/client.py", line 8, in hello
    websocket = await websockets.connect('ws://localhost:8765')
TypeError: __await__() returned a coroutine
```

**Comment by aaugustin** [comment]
Oops. Looking into this ASAP. Stick with 4.0 in the meantime.

The security issue which 5.0 fixes is unlikely to cause problems if you're connecting to servers that you trust.

**Comment by aaugustin** [comment]
I suspect 7da5f40a65fc6003b3d2457d042c07443a8256d1 was a mistake.

At worst I can revert it and figure out how to support Python 3.7 later.

**Comment by SimSimY** [comment]
What changes in 3.7? 
Pep 492 states that
> It is a TypeError if __await__ returns anything but an iterator.
And coroutines are not iterators

**Comment by aaugustin** [comment]
Before that change, tests crashed on Python 3.7 while they passed on 3.4 - 3.6. So there's definitely a change, although it doesn't appear to be documented (or at least I couldn't find it in the changelog).

I made this change based on by interpretation of the error message. However it seems wrong.

**Comment by cjerdonek** [comment]
What was the error using Python 3.7? There are ~11 hours to file a release blocker for Python 3.7, if there is a regression.

**Comment by aaugustin** [comment]
The error on Python 3.7 is:

```
$ python3.7 -m unittest --failfast
............................E
======================================================================
ERROR: test_iterate_on_messages (websockets.py36._test_client_server.AsyncIteratorTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/myk/Documents/dev/websockets/websockets/py36/_test_client_server.py", line 41, in test_iterate_on_messages
    server = self.loop.run_until_complete(start_server)
  File "/Users/myk/.pyenv/versions/3.7.0b2/lib/python3.7/asyncio/base_events.py", line 566, in run_until_complete
    return future.result()
  File "/Users/myk/.pyenv/versions/3.7.0b2/lib/python3.7/asyncio/tasks.py", line 570, in _wrap_awaitable
    return (yield from awaitable.__await__())
  File "/Users/myk/Documents/dev/websockets/websockets/server.py", line 742, in __await__
    server = yield from self._creating_server
TypeError: cannot 'yield from' a coroutine object in a non-coroutine generator

----------------------------------------------------------------------
Ran 29 tests in 0.008s

FAILED (errors=1)
sys:1: RuntimeWarning: coroutine 'BaseEventLoop.create_server' was never awaited
```

**Comment by aaugustin** [comment]
If I were to file a Python bug, I expect the conversation to be:

- Why do you want to import code that defines `__await__` on Python 3.4? It isn't going to be called
- To avoid duplicating 100 lines of code... Okay that isn't a very good reason, BRB, going to refactor

**Comment by lambdalisue** [comment]
Thanks for quick 5.0.1 release!

**Comment by aaugustin** [comment]
I just noticed that wheels weren't available because I had forgotten to push the 5.0.1 tag. They should be there shortly.
